### ISSUE TITLE
Admin panel: after giving away Owner permissions, "Set as Owner" still appears but silently fails

### ISSUE THREAD

**Issue by apenwarr** [tailscale/tailscale#948]
Steps:
- Log in as a domain owner
- Go to the login.tailscale.com/admin/users
- Click on the "..." icon next to any user other than yourself and pick "Set as Owner"
- Ownership changes in the display (your account becomes Admin, and the selected account becomes Owner), as expected
- Now click on the "..." icon next to any user other than the new owner (including yourself) and Pick "Set as Owner"

Expected:
- "Set as Owner" is not available, because only the Owner can give away ownership rights, and you are no longer the owner.

Actual:
- "Set as Owner" is available (bug 1)
- But if you pick it, it doesn't work (which is good, otherwise it would be a security hole)
- But it doesn't display any kind of message, even though the server should be returning an error in response to the RPC (bug 2). We should have a general error handler to report all such errors.

Bonus:
Bug 3: It was a little too easy to do "set as owner". There should be a big scary warning that this operation is irreversible so you'd better make sure you really mean it. There's also no explanation anywhere on the page about what "Owner" rights mean.

cc: @rosszurowski

**Comment by rosszurowski** [comment]
Good catch. Seems we're not refreshing the front-end user state after update, which is why the button appears but doesn't work.

> There's also no explanation anywhere on the page about what "Owner" rights mean.

Yup. Until we roll out a better team model this is going to remain a little confusing. Clarifying it might be a good low-priority bug to tackle too, but the real confusion lies deeper than our UI.

**Comment by rosszurowski** [comment]
Reframing this bug since the original issue has since been fixed: "set as owner" immediately disappears once the ownership has been transferred. And if this API call error, we now have a toaster in place to provide an error message to users. The one remaining piece that hasn't been fixed is providing a big, scary, explanatory warning of the destructive action that is about to happen.

**Comment by neilisaac** [comment]
This was not obvious because some services might allow multiple "owners" so it wasn't obvious (enough) to us that "set as owner" meant transfer ownership rather than add additional owner.

**Comment by alexwlchan** [comment]
We fixed this sometime between now and when this bug was written. Now if you try to change somebody to owner, we show you this confirmation dialog before changing ownership:

<img width="549" height="272" alt="Image" src="https://github.com/user-attachments/assets/4bebb70e-9093-42e5-ad98-3d8716981c60" />
